overlay_manager.js

```diff
@@ -449,67 +531,211 @@ class MemoryVolumeOverlay extends GenericSdfgOverlay {
+        let volume = undefined;
+        if (volume_string !== undefined)
+            volume = this.symbol_resolver.parse_symbol_expression(
+                volume_string,
+                symbol_map
+            );
+
+        edge.data.volume = volume;
+
+        if (volume !== undefined && volume > 0)
+                volume_values.push(volume);
+
+        return volume;
+    }
+
+    calculate_volume_graph(g, symbol_map, volume_values) {
+        let that = this;
+        g.nodes().forEach(v => {
+            let state = g.node(v);
+            let state_graph = state.data.graph;
+            if (state_graph) {
+                state_graph.edges().forEach(e => {
+                    let edge = state_graph.edge(e);
+                    if (edge instanceof Edge)
+                        that.calculate_volume_edge(
+                            edge,
+                            symbol_map,
+                            volume_values
+                        );
+                });
+
+                state_graph.nodes().forEach(v => {
+                    let node = state_graph.node(v);
+                    if (node instanceof NestedSDFG) {
+                        let nested_symbols_map = {};
+                        let mapping = node.data.node.attributes.symbol_mapping;
+                        // Translate the symbol mappings for the nested SDFG
+                        // based on the mapping described on the node.
+                        Object.keys(mapping).forEach((symbol) => {
+                            nested_symbols_map[symbol] =
+                                that.symbol_resolver.parse_symbol_expression(
+                                    mapping[symbol],
+                                    symbol_map
+                                );
+                        });
+                        // Merge in the parent mappings.
+                        Object.keys(symbol_map).forEach((symbol) => {
+                            if (!(symbol in nested_symbols_map))
+                                nested_symbols_map[symbol] = symbol_map[symbol];
+                        });
+
+                        that.calculate_volume_graph(
+                            node.data.graph,
+                            nested_symbols_map,
+                            volume_values
+                        );
+                    }
+                });
+            }
+        });
+    }
+
+    recalculate_volume_values(graph) {
+        this.badness_scale_center = 5;
+
+        let volume_values = [0];
+        this.calculate_volume_graph(
+            graph,
+            this.symbol_resolver.symbol_value_map,
+            volume_values
+        );
+
+        switch (this.badness_scale_method) {
+            case 'mean':
+                this.badness_scale_center = math.mean(volume_values);
+                break;
+            case 'mode':
+                this.badness_scale_center = math.mode(volume_values);
```

Comment:
There can be more than one mode in math.js!
https://mathjs.org/docs/reference/functions/mode.html

Reply:
Good catch, thank you!